src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.extension.file.common.api;
+
+import org.mule.functional.junit4.rules.ExpectedError;
+
+import java.nio.file.Paths;
+
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TestRule;
+
+/**
+ * A {@link TestRule} which provides tools and environment for reusing the same test regardless of the remote server type (FTP,
+ * SFTP, etc) or the authentication method
+ *
+ * @since 1.0
+ */
+public interface FileTestHarness extends TestRule {
```

Comment:
This class is not used on file connector, only on FTP and SFTP connectors.  Rename it

Reply:
No. This is supposed to be a generic abstraction for testing a connector which operates over a file system. Tomorrow we may do samba, google drive or dropbox and still want to have an implementation of this class.

Reply:
What is the expected result of calling getServerPort on a standard file system?

Reply:
for the particular case of a local file system, none. For any other case, it does make sense